app/javascript/data.js

```diff
@@ -0,0 +1,207 @@
+import createDebug from 'debug';
+import { onLoad } from './util';
+
+const debug = createDebug('ms:data');
+
+window.store = {};
+window.results = [];
```

Comment:
These two (and only these two) are global so that the user's custom script can access them, since that gets executed in a global scope.